mne/decoding/tests/test_time_gen.py

```diff
@@ -171,6 +171,25 @@ def test_generalization_across_time():
     assert_true(len(gat.test_times_['slices'][0]) == 15 ==
                 np.shape(gat.scores_)[1])
 
+    # Test score_mode
+    gat.score_mode = 'foo'
+    assert_raises(ValueError, gat.score, epochs)
+    gat.score_mode = 'per-fold'
+    scores = gat.score(epochs)
+    assert_array_equal(np.shape(scores), [15, 15, 5])
+    gat.score_mode = 'no-cv'
```

Comment:
Note that we actually have 3 approaches:
* keep the score of each fold (original request): `fold-wise`
* compute the score per fold and average: `mean-fold-wise`  (default)
* compute across samples: `sample-wise`